hummingbot/connector/budget_checker.py

```diff
@@ -1,35 +1,147 @@
 import typing
 from collections import defaultdict
 from copy import copy
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 from decimal import Decimal
-from typing import Dict, List, Optional
+from typing import Dict, List, Optional, Tuple
 
-from hummingbot.connector.utils import split_hb_trading_pair
+from hummingbot.connector.utils import split_hb_trading_pair, combine_to_hb_trading_pair
 from hummingbot.core.event.events import OrderType, TradeType
-from hummingbot.core.data_type.trade_fee import TradeFee
+from hummingbot.core.data_type.trade_fee import TradeFee, TradeFeePercentageApplication
 
 if typing.TYPE_CHECKING:  # avoid circular import problems
     from hummingbot.connector.exchange_base import ExchangeBase
 
 
+@dataclass
+class SizeEntry:
```

Comment:
`SizeEntry` is not a very clear name. I believe something like `TokenAmount` would be better.

Reply:
I like it; changed to `TokenAmount`